advanced_tutorials/on_demand_feature/README.md

```diff
@@ -0,0 +1,38 @@
+## Directory Structure
+
+├── LICENSE
+├── README.md          <- README explains this Python module to both developers and users.
+│
+├── notebooks          <- Jupyter notebooks. Naming convention is a number (for ordering),
+│   └── my_module      <- A symbolic link to the 'my_module' directory
+│                         On Linux/Mac: cd notebooks ; ln -s ../my_module .
+│
+├── requirements.txt   <- The requirements file for creating the Python environment. Install in a venv/conda environment.
+│                         `conda activate my_env`
+│                         my_env> `pip install -r requirements.txt`
+│
+├── setup.py           <- Make this project pip installable with `pip install -e`
+├── my_module          <- Source code for this project.
+│   ├── __init__.py    <- Makes a Python module
+│   │
+│   ├── pipelines      <- Feature pipelines, training pipelines, batch inference pipelines.
+│   │   │── feature_pipeline.py
+│   │   │── training_pipeline.py
+│   │   └── batch_inference_pipeline.py
+│   │
+│   ├── features       <- Python modules to turn raw data into features for use in both training and inference
+│   │   └── my_features.py
+│   │
+│   ├── transformations<- Python modules with model-specific transformation functions
+│   │   └── my_transformations.py
+│   │
+│   ├── tests          <- Pytest unit tests for feature logic
```

Comment:
And remove all the rest - tests, visualizations, etc - given we don't use them here.

Reply:
If you look at the README.md in your repo, it's not readable - i would just replace it with the following:

This example shows how to implement an on-demand feature in Hopsworsk - that is, a feature that is computed at request-time using application-supplied inputs for an online model .

We use a house-price estimated dataset/model to demonstrate how to implement an on-demand feature.
The on-demand feature here is the zipcode (or postcode) that is computed using longitude/latitude  parameters. In our online application,  longitude and latitude are provided as parameters to the application, and the same Python function used to calculate the zipcode in the feature pipeline is used to compute the zipcode in the Online Inference pipeline. This is achieved by implementing the on-demand features as a Python function in a Python module. We then ensure that the same version of the Python module is installed in both the feature and inference pipelines.

Reply:
In 
3_feature_view_td_modelling.ipynb
why use 'get_or_create_feature_group()' ? Is it not better with just 'get_feature_group' ?

Why create the TD with a Spark job and then read the files (it will take 5 minutes)?
Instead, just call
train_x, test_x, train_y, test_y = feature_view.train_test_split(test=0.2)

In the model saving part, 
why create the directory "housing_model" and then only save the pkl file to the registry?
model.save(pkl_file_name)

Replace this:
model_dir="housing_model"
if os.path.isdir(model_dir) == False:
    os.mkdir(model_dir)
pkl_file_name = model_dir + '/housing_model.pkl'

with this

pkl_file_name = 'housing_model.pkl'

Reply:
You have this line:
model = mr.get_model(name="housingmodel",version=1)
But if you run the notebook a 2nd time, you will create version 2 of the model, but you will always deploy version 1. 
One solution would be just to remove this line. If the user ran the notebook from top to bottom, the model object will already exist and be the version created.

Reply:
The most important thing to fix is:
you must include
feature_view.get_feature_vector( { passed : 'zipcode' : zipcode })

Reply:
You need to merge on-demand with precomputed features.

Reply:
Also, i don't think it's good practice to postfix feature group names with "_fg". Same for feature views with "_fv".  We don't postfix database tables  names with "_tb" or database names with "_db".